src/data/ProgramsD2Repository.ts

```diff
@@ -47,6 +64,31 @@ export class ProgramsD2Repository implements ProgramsRepository {
                 else throw new Error(JSON.stringify(err));
             });
     }
+
+    private getRealOrgUnits(ids: string[]) {
+        return Promise.allSettled(
+            ids.map(id =>
```

Comment:
Unbounded parallel promises. Not saying it's needed in this case, just as a note, if given N ids it will issue N requests to the instance.